src/inmanta/server/services/environment_metrics_service.py

```diff
@@ -321,6 +321,8 @@ def _divide_time_interval_in_time_windows(
         total_seconds_in_interval = (end_interval - start_interval).total_seconds()
         seconds_per_window = math.floor(total_seconds_in_interval / nb_time_windows)
         result = [end_interval - timedelta(seconds=seconds_per_window) * i for i in range(nb_time_windows)]
+        if round_timestamps_to_previous_hour:
```

Comment:
won't this lead to funny effects? 

e.g.
- if the window <1h we get intervals of size 0
- if the window is not divisible by 1h, we sometimes get additional intervals of size 0


I would instead, if we round:
1. round the interval boundaries to the hour
2. round up the seconds_per_window to hour (shortest interval is one hour)

the number of intervals may then be different, but at least it will behave regularly 
I would document that when round_timestamps_to_previous_hour is set, we can return less datapoints